ITensorGaussianMPS/src/gmps.jl

```diff
@@ -196,45 +489,81 @@ stored in a structure as a set of Givens rotations.
 
 The correlation matrix should be Hermitian, and will be treated as if it itensor
 in the algorithm.
+
+If `is_bcs`, the correlation matrix is assumed to be in interlaced format:
+Λ[2*i-1:2*i,2*j-1:2*j]=[[c_i c_j^dagger , c_i c_j ], [c_i^dagger c_j^dagger,c_i^dagger c_j]]
+Note that this may not be the standard choice in the literature, but it is internally
+consistent with the format of single-particle Hamiltonians and Slater determinants employed.
 """
+###Backward Compatibility
 function correlation_matrix_to_gmps(
-  Λ0::AbstractMatrix{ElT}; eigval_cutoff::Float64=1e-8, maxblocksize::Int=size(Λ0, 1)
-) where {ElT<:Number}
-  Λ = Hermitian(Λ0)
-  N = size(Λ, 1)
+  Λ0::AbstractMatrix; eigval_cutoff::Float64=1e-8, maxblocksize::Int=size(Λ0.data, 1)
+)
+  return correlation_matrix_to_gmps(
+    Λ0; eigval_cutoff=eigval_cutoff, minblocksize=1, maxblocksize=maxblocksize
+  )
+end
+
+function correlation_matrix_to_gmps(
+  Λ0::AbstractMatrix;
+  eigval_cutoff::Float64=1e-8,
+  minblocksize::Int=1,
+  maxblocksize::Int=size(Λ0.data, 1),
+)
+  return correlation_matrix_to_gmps(
+    ConservingNf(Λ0);
+    eigval_cutoff=eigval_cutoff,
+    minblocksize=minblocksize,
+    maxblocksize=maxblocksize,
+  )
+end
+
+function correlation_matrix_to_gmps(
+  Λ0::T;
+  eigval_cutoff::Float64=1e-8,
+  minblocksize::Int=1,
+  maxblocksize::Int=size(Λ0.data, 1),
+) where {T<:Union{Pairing,ConservingNf}}
+  Λ = T(Hermitian(copy(Λ0.data)))
+  ElT = eltype(Λ.data)
   V = Circuit{ElT}([])
-  ns = Vector{real(ElT)}(undef, N)
   err_tot = 0.0
-  for i in 1:N
-    blocksize = 0
-    n = 0.0
+  Λ = check_pairing_correlations(Λ)
+  N = size(Λ.data, 1)
+  calctype = typeof(Λ)
+  factor = calctype <: Pairing ? 2 : 1
+  ns = calctype(Vector{real(ElT)}(undef, N))
+  for i in 1:div(N, factor)
     err = 0.0
-    p = Int[]
-    uB = 0.0
-    for blocksize in 1:maxblocksize
-      j = min(i + blocksize, N)
-      ΛB = @view Λ[i:j, i:j]
-      nB, uB = eigen(Hermitian(ΛB))
-      p = sortperm(nB; by=entropy)
-      n = nB[p[1]]
-      err = min(n, 1 - n)
-      err ≤ eigval_cutoff && break
+    v, nB, err = get_subblock(
+      Λ,
+      i;
+      eigval_cutoff=eigval_cutoff,
+      minblocksize=minblocksize,
+      maxblocksize=maxblocksize,
+    )
+    ns, to_break = process_populations!(ns, nB, v, i)
```

Comment:
I found the logic of these two functions hard to follow.

It seems cleaner to have an interface like this:
```julia
    v, n, err = isolate_subblock_eigs(Λ, i; eigval_cutoff, minblocksize, maxblocksize) # Outputs only the isolated/disentangled eigenvector(s)/eigenvalue(s)
    set_occupations!(ns, i, n)
    stop_gmp_sweep([Whatever is needed to determine when to stop the sweep])
```